### DISCUSSION TITLE
Opening shadcn dialog resets scroll to top

### DISCUSSION THREAD

**Discussion by WillSmithTE** [epicweb-dev/epic-stack#565] [Q&A]
I'm losing scroll position after adding the shadcn dialog and opening it, should this be expected to work? Could be related to #346 

**Steps**
1. `npx create-epic-app@latest`
2. `npm install lucide-react` (for shadcn dialog)
3. `npx shadcn-ui@latest add dialog`
4. Add the dialog example from the shadcn docs to a point "below the fold", [like this](https://github.com/WillSmithTE/dialog-scroll/blob/main/app/routes/_marketing%2B/index.tsx#L89)
5. Click "open"

**Expected**
Dialog opens and the scroll position is preserved

**Actual**
Dialog opens but scroll position resets to the top

In this repo I've done that, the only change to the initial template are the shadcn dialog bits: https://github.com/WillSmithTE/dialog-scroll

https://github.com/epicweb-dev/epic-stack/assets/26863411/f824a871-8b58-4d22-b945-796496a8f97e

**Comment by kentcdodds** [comment]
I'm not certain what is triggering the scroll reset. Can you determine what's triggering the reset?

**Reply by WillSmithTE** [reply]
It took me 2 hours just to reduce it down to this repro, so I'm really not sure, the only I've found by trial and error is setting `modal` to false won't reset the scroll (but won't have the overlay).

I'd need to dig further into radix and the shadcn styles

**Reply by eckhartt** [reply]
I'm also experiencing this with shadcn Select, and I've noticed it also happens with the User dropdown menu. 

I think it might relate to this being injected into the head div

<img width="464" alt="image" src="https://github.com/epicweb-dev/epic-stack/assets/12352826/1262b637-0d63-4e5c-98ed-53c3f5262c15">


edit:

Making the below change at the top of Document() in app/root.tsx seems to have fixed the issue but I don't know enough to understand the implications 

```javascript
		// <html lang="en" className={`${theme} h-full overflow-x-hidden`}>
		<html lang="en" className={`${theme} h-full`}>
```

**Reply by jacobparis** [reply]
I came to the same conclusion when working through this, it seems shadcn works better with body scrolling than html scrolling, 

https://github.com/jacobparis/epic-radial/commit/e3397c98d6a9b1d33c375768fc31a5e797065dd9

**Reply by lewisblackburn** [reply]
I've been stuck on this for a while, thanks for looking into this further guys. Removing `overflow-x-hidden` seems to fix the issue with no repercussions for me.

**Comment by jamestrenda** [comment]
I ran into this issue with DropdownMenus. It's because the epic-stack's app is styled to have a [fixed height](https://github.com/epicweb-dev/epic-stack/blob/b4e118ddef884a4e3a3b94da0eb2b02d91a35506/app/root.tsx#L237). My app does not need a fixed height. Swapping `h-screen` for `min-h-screen` is what I did.

**Reply by jamestrenda** [reply]
To be fair to Kent, the issue is technically not because of the fixed height. It's a combination of the fixed height of the app and the styles that are injected by shadcn when the modal is displayed. I would expect the components to work with or without a fixed height, but it was just easier to change the class name for my use case.

**Reply by wallacepreston** [reply]
This saved me!  Thank you @jamestrenda !

**Comment by CodingAleCR** [comment]
With Dropdown Menu & Select, I saw that there are issues within the primitives in Radix UI that cause this. In fact I had to migrate to using Dropdown Menu instead of select because of it.

Anyways, in those scenarios, and specifically with Dropdown Menu, the way to fix it was to set `modal="false` on `DropdownMenu.Root` for it to avoid resetting scroll to the top. It might be the same in this case (maybe?).

**Comment by chaikin387** [comment]
В моем случае диалоговое окно открывалось через Link на карточке товара и тоже был скролл в начало страницы, при закрытии скролилось обратно к карточке. Решение нашлось путем добавления к Link scroll={false}.

**Comment by ArthurYdalgo** [comment]
In case anyone is also having this issue and none of the suggestions above fixed it for you: 

I had `h-full` in both my `<html ... >` and `<body ... >` tags... removing this class from the `<body ... >` fixed it for me. 

edit: typo
